bitbots_dynamic_kick/include/bitbots_dynamic_kick/kick_engine.h

```diff
@@ -180,6 +185,8 @@ class KickEngine : public bitbots_splines::AbstractEngine<KickGoals, KickPositio
    * @param ball_position Position of the ball
    * @param kick_direction Direction in which to kick the ball
    * @return pair of (transformed_pose, transformed_direction)
+   *
+   * @throws tf2::TransformException when goal cannot be transformed into support_foot_frame
    */
   std::optional<std::pair<geometry_msgs::Point, geometry_msgs::Quaternion>> transformGoal(
```

Comment:
This should either return an optional or throw an exception. If we want to do it with exception, please change the return type.